### PR TITLE
Export `MockType` for type annotations

### LINKED ISSUE #414: Allow for easier import of `mocker.Mock | mocker.MagicMock` types
I want to be able to import `Mock` and `MagicMock` types for type annotations.

I don't want to have to import from `unittest.mock` because I am enforcing the use of `pytest-mock` [by banning the import](https://docs.astral.sh/ruff/rules/banned-api/) of `unittest.mock`.

```python
from typing import TYPE_CHECKING

import pytest

if TYPE_CHECKING:
  from pytest_mock import Mock, MagicMock, MockerFixture

@pytest.fixtue
def my_mock(mocker: MockerFixture) -> Mock:
   return mocker.Mock()


@pytest.fixtue
def my_magic_mock(mocker: MockerFixture) -> MagicMock:
   return mocker.MagicMock()


def test_something(my_mock: Mock):
   ...
```

I see 2 different approaches here.
1. "Export" `unittest.mock` `Mock` + `MagicMock` in the top-level [`__init__` file](https://github.com/pytest-dev/pytest-mock/blob/main/src/pytest_mock/__init__.py)
2. Add a `pytest_mock.types` subpackage. This could also include things like `ANY` which would otherwise need to be imported from `unittest.mock`.

`1` could be an issue because it might encourage users to use the `Mock`/`Mocker` as more than just type-annotations.

### PR DESCRIPTION
**Pull request by Kilo59**
- Resolves #414

### CODE REVIEW

File: src/pytest_mock/__init__.py [hunk 1]
```diff
@@ -13,6 +14,7 @@
 __all__ = [
     "MockerFixture",
     "MockFixture",
+    "MockType",
```
[the remark this answers is not in the document]
Comment (Kilo59) [reply src/pytest_mock/__init__.py:19]: Done

[the remark this answers is not in the document]
Comment (imnotjames) [reply src/pytest_mock/__init__.py:19]: 🎉 

I'm not a maintainer to `pytest-mock` but it just so happened that I thought to check github issues for this & you had made an awesome change that solved issue.

Thank you!

**Comment by nicoddemus** [comment]
Sorry @Kilo59 I've missed the notification for this PR, I will take a look in a few days. 👍

**Comment by nicoddemus** [comment]
Actually found some time to review it and is pretty simple. Thanks!

**Pull request merged**